src/modules/organisation-permissions/helpers/custom-validator.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+import { PermissionCategory } from './PermissionCategory';
+
+@ValidatorConstraint({ async: false })
+class IsPermissionListValidConstraint implements ValidatorConstraintInterface {
+  validate(permissionList: any) {
+    if (typeof permissionList !== 'object' || permissionList === null) {
+      return false;
+    }
+
+    const validCategories = Object.values(PermissionCategory);
+
+    for (const [key, value] of Object.entries(permissionList)) {
+      if (!validCategories.includes(key as PermissionCategory)) {
+        return false;
+      }
+      if (typeof value !== 'boolean') {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  defaultMessage() {
```

Comment:
Use a constant variable for this

Reply:
Hey bro I am not sure what I should do here, the default message is meant to be a function that is returned when validation fails.I can only be done in the function